### DISCUSSION TITLE
Console log spamming

### DISCUSSION THREAD

**Discussion by Hiekki** [WFCD/warframe-worldstate-parser#302] [Q&A] [answered]
I recently looked into this API for a friend to make a discord bot and it constantly spams the console with:
```
No defined kuva data, skipping data
No outpost data, skipping
```

Is there an easy way to silence this?
```
const WorldState = require('warframe-worldstate-parser');
const request = require('request-promise');

const worldStateData = await request('http://content.warframe.com/dynamic/worldState.php');

//this is the line where it posts
const ws = new WorldState(worldStateData); 
```

Any help would be appreciated as this is rather annoying and I'm not sure what to do to stop it. Thank you!

**Comment by TobiTenno** [comment, marked as answer]
if you implement a dummy logger (we use console as a reference for what logging we do), you can turn all of them off or use a logger that lets you set the "log level" like the one we use in genesis or something like winston.

alternatively, if you don't want to parse the data yourself, you can fetch from [our API](https://api.warframestat.us). ([Docs](https://docs.warframestat.us))

**Reply by Hiekki** [reply]
i have tried to completely stop the console.log entirely and it still posts, sadly.

```
console.log = function () {};
```

edit: i also tested this with another command and it works perfectly. just not with this

**Reply by TobiTenno** [reply]
yeah, we don't use console.log, so that wouldn't work. you can see the [dummy logger impl in our test code](https://github.com/WFCD/warframe-worldstate-parser/blob/master/test/integration/integration.spec.js#L10-L15) if you want.

```js
const logger = {
  error: () => {},
  debug: () => {},
  info: () => {},
  log: () => {},
};
```

**Reply by TobiTenno** [reply]
I use this day in and day out with winston, setting up a really [basic logger](https://github.com/WFCD/worldstate-emitter/blob/master/utilities/index.js#L3-L21)

```js
require('colors');
const { transports, createLogger, format } = require('winston');


const {
  combine, label, printf, colorize,
} = format;


/* Logger setup */
const transport = new transports.Console({ colorize: true });
const logFormat = printf((info) => `[${info.label}] ${info.level}: ${info.message}`);
const logger = createLogger({
  format: combine(
    colorize(),
    label({ label: 'WS'.brightBlue }),
    logFormat,
  ),
  transports: [transport],
});
logger.level = process.env.LOG_LEVEL || 'error';
```

**Reply by Hiekki** [reply]
i could kiss you, in the biggest homo way possible. that worked <3

thank you!!
